[refactor] Remove unnecessary checks in InsertQuantDeQuantImpl

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25403 [clang-format] torch/jit/passes/quantization.{h,cpp} and torch/jit/init.cpp
* **#25370 [refactor] Remove unnecessary checks in InsertQuantDeQuantImpl**
* #25281 [quant][graphmode] Skip inserting observers for Tensors inside fused op

Summary:
Removes some checking code that is copied from insert_observers pass

Test Plan:
python test/test_jit.py 'TestJit.test_insert_quant_dequant'

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106633](https://our.internmc.facebook.com/intern/diff/D17106633)

jerryzh168 committed:
[refactor] Remove unnecessary checks in InsertQuantDeQuantImpl

Summary:
Removes some checking code that is copied from insert_observers pass

Test Plan:
python test/test_jit.py 'TestJit.test_insert_quant_dequant'

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

ZolotukhinM reviewed (commented):
I think we can clean it up more, please see inline.

jerryzh168 committed:
Update on "[refactor] Remove unnecessary checks in InsertQuantDeQuantImpl"


Summary:
Removes some checking code that is copied from insert_observers pass

Test Plan:
python test/test_jit.py 'TestJit.test_insert_quant_dequant'

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106633](https://our.internmc.facebook.com/intern/diff/D17106633)

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -521,23 +511,19 @@ void InsertQuantDeQuantImpl(
-      } else {
-        qh.quantizeTensor(v);
-      }
+    auto child_module = qh.findChildModuleToQuantize(v);
```
sure I can move this to first loop

jerryzh168 committed:
Update on "[refactor] Remove unnecessary checks in InsertQuantDeQuantImpl"


Summary:
Removes some checking code that is copied from insert_observers pass

Test Plan:
python test/test_jit.py 'TestJit.test_insert_quant_dequant'

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106633](https://our.internmc.facebook.com/intern/diff/D17106633)

ZolotukhinM reviewed (approved):
Yep, looks better now, thank you!

jerryzh168 committed:
Update on "[refactor] Remove unnecessary checks in InsertQuantDeQuantImpl"


Summary:
Removes some checking code that is copied from insert_observers pass

Test Plan:
python test/test_jit.py 'TestJit.test_insert_quant_dequant'

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17106633](https://our.internmc.facebook.com/intern/diff/D17106633)
